tiledb/sm/query/legacy/reader.cc

```diff
@@ -336,10 +336,9 @@ Status Reader::load_initial_data() {
   }
 
   // Load delete conditions.
-  auto&& [st, conditions, update_values] =
+  auto&& [conditions, update_values] =
       array_->load_delete_and_update_conditions();
-  RETURN_CANCEL_OR_ERROR(st);
```

Comment:
We can't remove any of these calls to `RETURN_CANCEL_OR_ERROR` until we have a replacement that cancels the query. Not having such a replacement blocks a change to the return value of `load_delete_and_update_conditions`.

Even with a replacement, the call will need to be in a `try` block whose `catch` contains a statement to cancel the query.